init.lua

```diff
@@ -58,59 +63,69 @@ minetest.register_chatcommand("watch", {
 	privs = {watch = true},
 	func = function(name_watcher, name_target)
 		if name_watcher == name_target then
-			return true, "You may not watch yourself"
+			return true, "You may not watch yourself."
 		end
 
 		local target = minetest.get_player_by_name(name_target)
 
 		if not target then
-			return true, "Unknown target player name"
+			return true, 'Invalid target name "' .. name_target .. '"'
 		end
 
 		-- avoid infinite loops
-		if original_pos[name_target] then
-			return true, name_target .. " is already watching a player."
+		if original_state[name_target] then
+			return true, '"' .. name_target .. '" is watching "'
+				.. original_state[name_target].target .. '". You may not watch a watcher.'
 		end
 
 		local watcher = minetest.get_player_by_name(name_watcher)
-		local privs_watcher = minetest.get_player_privs(name_watcher)
-
 		if player_api.player_attached[name_watcher] == true then
-			unwatching(name_watcher)
+			detach(name_watcher)
 		end
-		original_pos[name_watcher] = watcher:get_pos()
-
-		player_api.player_attached[name_watcher] = true
-		watcher:set_attach(target, "", vector.new(0, -5, -20), vector.new())
-		watcher:set_eye_offset(vector.new(0, -5, -20), vector.new())
-		watcher:set_nametag_attributes({color = {a = 0}})
-
-		toggle_hud_flags(watcher, true)
 
+		-- back up some attributes
+		local properties = watcher:get_properties()
+		original_state[name_watcher] = {
+			collisionbox = table.copy(properties.collisionbox),
+			hud_flags = table.copy(watcher:hud_get_flags()),
+			nametag = table.copy(watcher:get_nametag_attributes()),
```

Comment:
Is this backup necessary? Couldn't you set `a=0` to turn the text off to allow changes while the player is watching?

Reply:
I did not test that yet. Went with work-first aproach that would cover any future changes too.
If that's not a good approach, I'll change it in 'invite' branch.